File: 1. Study/0606 - 자료구조(스택)/mojaeya/2_괄호문자제거.js
let str = "(A(BC)D)EF(G(H)(IJ)K)LM(N)";
console.log(solution(str));

function solution(str) {
  let answer;
  let stack = [];
  for (let x of str) {
    if (x === ")") {
      while (stack.pop() !== "("); // (가 나올 때까지 모두 pop
    } else stack.push(x); // (와 문자는 모두 push
    // console.log(stack);
  }
  answer = stack.join("");
  return answer;
}

File: 1. Study/0606 - 자료구조(스택)/kimshin/2. 괄호문자제거.py
def sol(s):
    answer = ''
    stack = []

    for x in s:
        if x == ')':
            while stack.pop() !='(':
                continue
        else:
            stack.append(x)
    answer = ''.join(stack)
    return answer


a = "(A(BC)D)EF(G(H)(IJ)K)LM(N)";
print(sol(a))